src/main/java/org/apache/commons/compress/compressors/gzip/GzipCompressorOutputStream.java

```diff
@@ -185,14 +196,26 @@ private void writeHeader(final GzipParameters parameters) throws IOException {
             buffer.put((byte) 0);
         }
         buffer.put((byte) parameters.getOperatingSystem());
+
         out.write(buffer.array());
+        crc.update(buffer.array());
         if (extra != null) {
             out.write(extra.length & 0xff); // little endian
             out.write(extra.length >>> 8 & 0xff);
             out.write(extra);
+            crc.update(extra.length & 0xff);
+            crc.update(extra.length >>> 8 & 0xff);
+            crc.update(extra);
         }
         writeC(fileName, parameters.getFileNameCharset());
         writeC(comment, parameters.getFileNameCharset());
+
```

Comment:
Remove extra whitespace, if you need to point something out, use an `// inline comment`.

Reply:
nitpicking.